fetch.bs

```diff
@@ -3237,7 +3250,102 @@ through TLS using ALPN. The protocol cannot be spoofed through HTTP requests in
+
+<h4 id=https-upgrades-upgrade>HTTPS upgrade algorithm</h4>
+
+To <dfn>upgrade an HTTP request</dfn> given a <a for=/>request</a> <var>request</var>, run these
+steps:
+
+<ol>
+  <li>
+    <p>If one or more of the following conditions are met, return:
+    <ul>
+      <li><p><var>request</var>'s <a for="request">destination</a> is not "<code>document</code>"
+
+      <li><p><var>request</var>'s <a for="request">method</a> is not "<code>GET</code>"
+
+      <li><p><var>request</var>'s <a for="request">URL</a>'s <a for="url">scheme</a> is not
+      "<code>http</code>"
+
+      <li><p><var>request</var>'s <a for="request">URL</a>'s <a for="url">host</a> is exempted from
+      upgrades in an <a>implementation-defined</a> way.
+
+      <p class="example" id="example-https-upgrades-exempted-hosts">If <a for=url>host</a> is a
+      non-registrable or non-assignable domain name such as .local or an IP address that falls in a
+      range reserved for non-publicly routable networks, the implementation might return without
+      modifying <var>request</var>.
+    </ul>
+  </li>
+
+  <li><p>If <var>request</var>'s <code>https-upgrade-fallback-url</code> is non-null, clear the
+  <code>https-upgrade</code> and <code>https-upgrade-fallback-url</code> flags and return.
+
+  <p class="note">This is a fallback request that shouldn't be upgraded again.
+
+  <li><p>Otherwise, set the following fields:
+    <ul>
+      <li>Set <code>https-upgrade-fallback-url</code> to <var>request</var>'s <a for="request">URL</a>.
+      <li>Set <var>request</var>'s <a for="request">URL</a>'s <a for="url">scheme</a> to "<code>https</code>".
+      <li>Set <code>https-upgrade</code> flag to true.
+    </ul>
+  </li>
+</ol>
+
+<h4 id=https-upgrades-fallback>Fallback algorithm</h4>
+
+To run <dfn>HTTPS upgrade fallback</dfn> given a <a for=/>request</a>
+<var>request</var> and <a for=/>response</a> <var>response</var>, run these steps:
+
+<ol>
+  <li><p>If <var>request</var>'s <code>https-upgrade</code> flag is not set, return
+  <var>response</var>.
+
+  <li>
+    <p>If <var>response</var> is a network error, run the following steps:
+    <p class="note">This means that the upgrade failed and initiates a fallback load.
+     <ol>
+      <li><p>Let <var>fallbackResponse</var> be a new <a for=/>Response</a> with its <code>Location</code> header set to
+      <code>https-upgrade-fallback-url</code>, and its <a for="response">status</a> set to 307.
+      <li>Return <var>fallbackResponse</var>.
+     </ol>
+
+  <li>
+    <p class="note">The upgrade was successful.
+    <p>Return <var>response</var>.
+</ol>
+
+<p class=note>
+User agents can implement a fast-fallback path by canceling slow fetches on upgraded
+requests, in order to quickly initiate a fallback http load.
+
+</div>
+
+<h4 id=http-upgrades-examples>Examples</h4>
+
+<div id=example-https-upgrade-good-https class=example>
+ <p>
+ <code>a.com</code> serves both <code>http://a.com</code> and <code>https://a.com</code>.
```

Comment:
I'm not sure why this is on a newline. A `<p>` should have its contents on the same line (up to 100 columns). This applies various times.

Reply:
Done